File: swift/IsbnVerifier.swift
func isValid(_ i: String) -> Bool {
    let isbn = i.filter { "0123456789X".contains($0) }.flatMap { $0 == "X" ? 10 : Int(String($0)) }
    
    guard isbn.count == 10 && !isbn.prefix(9).contains(10) else { return false }

    return (0..<10).map { isbn[$0] * (10 - $0) }.reduce(0, +) % 11 == 0
}

File: clojure/isbn_verifier.clj
(ns isbn-verifier
  (:require [clojure.string :as string]))

(defn isbn?
  "Check if a given ISBN-10 is valid."
  [isbn]
  (if (string/includes? (subs (string/reverse isbn) 1) "X")
    false
    (let [lst     (string/split isbn #"")
          valid   (filter #(string/includes? "0123456789X" (str %)) lst)
          isbn'   (map #(Integer/parseInt %) (replace {"X" "10"} valid))
          r       (range 0 (count isbn'))
          result  (mod (reduce + (for [i r] (* (nth isbn' i) (- 10 i)))) 11)]
      (== result 0))))
